src/main/java/org/betonquest/betonquest/BetonQuest.java

```diff
@@ -1547,6 +1548,10 @@ public String getVariableValue(final String packName, final String name, final S
                 log.warn(pack, "Could not resolve variable '" + name + "'.");
                 return "";
             }
+            if ((playerID == null || playerID.trim().isEmpty()) && !var.isStaticness()) {
```

Comment:
[`isBlank()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#isBlank()) should solve the PMD violation while being a lot cleaner.
```suggestion
            if ((playerID == null || playerID.isBlank()) && !var.isStaticness()) {
```